src/utils/network-monitor.ts

```diff
@@ -677,6 +677,7 @@ export class NetworkMonitor {
           throw new Error('Unsupported RPC provider protocol -> ' + protocol)
       }
 
+      this.walletNonces[network] = -1
```

Comment:
why negative one?

Reply:
This is used as an indicator to logic that needs wallet nonce for transaction that the nonce was not loaded and needs to be queried for. Afterwards, nonces are managed internally and allows for our execute transaction to constantly call the RPC for tx submission with updated gas pricing, allowing to override pending tx if price increased.